### PR TITLE
[Cleanup] Updates ShopifyError field names and the callback ordering for the native/web pay APIs

### PR DESCRIPTION
**Pull request by sleroux**

### CODE REVIEW

**Comment by sleroux** [comment]
I'll add the callback ordering change to this in a bit.

**Comment by mikkoh** [comment]
@sleroux note that CI is failing most likely some tests need to also be updated with this change.

**Comment by sleroux** [comment]
Updated the tests and added the callback ordering change as well to this PR.

**Comment by sleroux** [comment]
Accidentally closed the PR. Reopened now!

File: scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.cs.erb [hunk 1]
```diff
@@ -61,7 +61,7 @@ namespace <%= namespace %>.SDK.iOS {
         /// <param name="success">Delegate method that will be notified upon a successful payment</param>
         /// <param name="failure">Delegate method that will be notified upon a failure during the checkout process</param>
         /// <param name="cancelled">Delegate method that will be notified upon a cancellation during the checkout process</param>
-        public void Checkout(string key, CheckoutSuccessCallback success, CheckoutFailureCallback failure, CheckoutCancelCallback cancelled) {
+        public void Checkout(string key, CheckoutSuccessCallback success, CheckoutCancelCallback cancelled, CheckoutFailureCallback failure) {
```
Comment (mikkoh) [inline scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.cs.erb]: Should we just delete this method to not cause confusion?

Comment (sleroux) [inline scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.cs.erb]: Wouldn't hurt to make it more clear. I'll remove for now.

File: scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.cs.erb [hunk 2]
```diff
@@ -50,20 +50,6 @@ namespace <%= namespace %>.SDK.iOS {
...
-        /// Starts the process of making a payment through Apple Pay.
-        /// </summary>
-        /// <remarks>
-        ///  Displays a payment interface to the user based on the contents of the Cart
-        /// </remarks>
-        /// <param name="key">Merchant ID for Apple Pay from the Apple Developer Portal</param>
-        /// <param name="success">Delegate method that will be notified upon a successful payment</param>
-        /// <param name="failure">Delegate method that will be notified upon a failure during the checkout process</param>
-        /// <param name="cancelled">Delegate method that will be notified upon a cancellation during the checkout process</param>
-        public void Checkout(string key, CheckoutSuccessCallback success, CheckoutFailureCallback failure, CheckoutCancelCallback cancelled) {
```
Comment (dengjeffrey) [inline scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.cs.erb]: Why was this method removed?

File: scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.cs.erb (hunk 1 above)
Comment (sleroux) [inline scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.cs.erb]: Actually if we remove this method the class won't adhere to the interface which causes problems above. Putting it back...

Comment (mikkoh) [inline scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.cs.erb]: Yep just throw a `NotImplementedException`

**Pull request merged**